packages/cli/src/credentials/oauth2Credential.api.ts

```diff
@@ -162,6 +163,18 @@ oauth2CredentialController.get(
 			returnUri = `${get(oauthCredentials, 'authUrl', '') as string}?${percentEncoded}`;
 		}
 
+		if (oauthCredentials.grantType === 'pkce') {
+			let { code_verifier, code_challenge } = pkceChallenge();
+			// Twitter OAuth2 doesn't allow ~ in the code_verifier
+			if (code_verifier.includes('~')) {
```

Comment:
instead of looping, we can patch the library to remove the `~`
https://github.com/crouchcd/pkce-challenge/blob/master/src/index.ts#L23